docs/features/incremental-generators.cookbook.md

```diff
@@ -636,7 +636,248 @@ TODO: https://github.com/dotnet/roslyn/issues/72149
+                ClassDeclarationSyntax classDeclarationSyntax = (ClassDeclarationSyntax)ctx.TargetNode;
+                
+                string className = classDeclarationSyntax.Identifier.ValueText;
+                string classNameSpace = GetClassNameSpace(classDeclarationSyntax.Parent);
+                InterfaceModel[] interfaces = GetInterfaceModels(ctx.SemanticModel.Compilation, classDeclarationSyntax);
+
+                return new Model(
+                    className,
+                    classNameSpace,
+                    interfaces
+                    );
+            }).Where(m => m is not null);
+
+        context.RegisterSourceOutput(provider.Collect(), Execute);
+    }
+    private static string GetClassNameSpace(SyntaxNode? parent)
+    {
+        return parent is NamespaceDeclarationSyntax namespaceDeclarationSyntax
+            ? namespaceDeclarationSyntax.Name.ToString()
+            : parent is FileScopedNamespaceDeclarationSyntax fileScopedNamespaceDeclarationSyntax
+            ? fileScopedNamespaceDeclarationSyntax.Name.ToString()
+            : AutoImplementAttributeNameSpace;
+    }
+
+    private static InterfaceModel[] GetInterfaceModels(Compilation compilation, ClassDeclarationSyntax classDeclarationSyntax)
+    {
+        //Get class usings in order to build full interface name if needed
+        string[] usings = GetUsings(classDeclarationSyntax.SyntaxTree.GetRoot());
+        string[] interfacesNames = GetInterfacesNames(classDeclarationSyntax.AttributeLists);
+
+        List<InterfaceModel> ret = [];
+        foreach (string interfaceName in interfacesNames)
+        {
+            INamedTypeSymbol? interfaceSymbol = GetInterfaceSymbol(compilation, usings, interfaceName);
+            if (interfaceSymbol is null)
+                continue;
+
+            //Get interface usings, in order to be sure to have the needed usings for the properties that will be generated
+            string[] interfaceUsings = interfaceSymbol
+                .DeclaringSyntaxReferences
+                .FirstOrDefault()?
+                .GetSyntax()
+                .SyntaxTree
+                .GetRoot()
+                .DescendantNodes()
+                .OfType<UsingDirectiveSyntax>()
+                .Select(x => x.ToString())
+                .ToArray() ?? [];
+
+            IPropertySymbol[] interfaceProperties = interfaceSymbol
+                .GetMembers()
+                .OfType<IPropertySymbol>()
+                .ToArray();
+
+            InterfacePropertyModel[] propertyModels = interfaceProperties
+                .Select(interfaceProperty =>
+                {
+                    /*Using "interfaceProperty.Type" instead of "interfaceProperty.Type.Name" in order to not have error in specific cases.
+                    Es. the type "int" has Name "Int32", but writing "Int32" casue compilation error if we are not using "System" namespace*/
+                    string type = interfaceProperty.Type.ToString();
```

Comment:
Use an appropriate `ToDisplayString()` instead. The same holds true for everywhere else you are converting type symbols to strings. You likely want `SymbolDisplayFormat.FullyQualifiedFormat`.